src/apis/githubAPI.ts

```diff
@@ -99,8 +109,105 @@ class GithubAPI {
         }
     }
 
-    async searchCommit(commitHash: string, repository: Repository = SERENITY_REPOSITORY) {
+    async getUser(author: string) {
+        try {
+            let username = author;
+            if (author.includes("@")) {
+                const search = await this.octokit.search.users({
+                    q: `${author.includes("@") ? `${author} in:email` : `user:${author}`}:`,
+                    per_page: 1,
+                });
+
+                if (search.data.total_count <= 0)
+                    throw new Error(`A GitHub user with the primary email ${author} was not found`);
+
+                username = search.data.items[0].login;
+            }
+
+            const results = await this.octokit.users.getByUsername({
+                username,
+            });
+
+            return results.data;
+        } catch (e: any) {
+            if (e.status === 404) return null;
+
+            console.trace(e);
+            throw e;
+        }
+    }
+
+    async getCommitsCount(
+        author: string,
+        repository: Repository = SERENITY_REPOSITORY,
+        attempt = 0
+    ): Promise<number | undefined> {
+        try {
+            const results = await this.octokit.search.commits({
+                q: `${author.includes("@") ? "author-email" : "author"}:${author}+repo:${
+                    repository.owner
+                }/${repository.name}`,
+                sort: "committer-date",
+                order: "desc",
+                per_page: 1,
+            });
+
+            return results.data?.total_count;
+        } catch (e: any) {
+            // TODO: Generalize rate-limiting retrying?
+            if (e.status === 403 && attempt < 5) {
+                const resetTimestamp = e.response?.headers?.["x-ratelimit-reset"]
+                    ? Number.parseInt(e.response?.headers?.["x-ratelimit-reset"], 10) * 1000
+                    : 0;
+
+                // Use a 60 seconds timeout as a fallback.
+                const timeout =
+                    resetTimestamp > Date.now()
+                        ? resetTimestamp - Date.now() + 1000 * 5
+                        : 1000 * 60;
+
+                console.info(
+                    `github(${repository.owner}/${repository.name}): Got rate limited for ${
+                        timeout / 1000
+                    } second(s)`
+                );
+
+                return new Promise((resolve, reject) => {
+                    setTimeout(async () => {
+                        try {
+                            const results = await this.getCommitsCount(
+                                author,
+                                repository,
+                                ++attempt
+                            );
+                            resolve(results);
+                        } catch (e) {
+                            reject(e);
+                        }
+                    }, timeout);
+                });
```

Comment:
can we do something like

```suggestion
                await new Promise(resolve => setTimeout(resolve, timeout));
                
                // try { ...
```

instead?

Reply:
Done 🤠